pytensor/tensor/extra_ops.py

```diff
@@ -1554,9 +1560,26 @@ def broadcast_shape_iter(
                 result_dims.append(first_length)
                 continue
 
-            # Add assert that all remaining shapes are equal
-            condition = pt_all([pt_eq(first_length, other) for other in other_lengths])
-            result_dims.append(_broadcast_assert(first_length, condition))
+            if not allow_runtime_broadcast:
+                # Add assert that all remaining shapes are equal
+                condition = pt_all(
+                    [pt_eq(first_length, other) for other in other_lengths]
+                )
+                result_dims.append(_broadcast_assert(first_length, condition))
+            else:
+                lengths = as_tensor_variable((first_length, *other_lengths))
+                runtime_broadcastable = pt_eq(lengths, one)
+                result_dim = pt_abs(
+                    pt_max(switch(runtime_broadcastable, -one, lengths))
+                )
+                condition = pt_all(
+                    switch(
```

Comment:
Maybe use an `or` instead of the switch?

Reply:
I thought about that, but since this code has a such a narrow focus now I went with easy of read